node/pkg/common/nodekey_test.go

```diff
@@ -0,0 +1,53 @@
+package common
+
+import (
+	"fmt"
+	"github.com/libp2p/go-libp2p-core/crypto"
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/zap"
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"testing"
+)
+
+func TestGetOrCreateNodeKeyWithNewPath(t *testing.T) {
+	// Get a non-existing temp file path to write auto-generated privKey to
+	path := "/tmp/node_key_test_" + fmt.Sprint(rand.Int())
+
+	logger, _ := zap.NewProduction()
+	privKey1, _ := GetOrCreateNodeKey(logger, path)
+	assert.NotNil(t, privKey1)
+
+	// Re-read the generated privKey file back into memory
+	b, _ := ioutil.ReadFile(path)
+	privKey2, _ := crypto.UnmarshalPrivateKey(b)
+
+	// Make sure we got the same key
+	assert.Equal(t, privKey1, privKey2)
+
+	// Manual cleanup
+	os.Remove(path)
```

Comment:
probably want to defer this like you did in the below test.

Reply:
good call, fixed.